### PR TITLE
Add jobs for the tests, client-go, and common-files repos.

### PR DESCRIPTION
**Pull request by geeknoid**

### CODE REVIEW

**Review by howardjohn — APPROVED** [review]
why are we not using the config generation for these? is the only thing the builder image?

File: prow/cluster/jobs/istio/client-go/istio-client-go.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+    - name: client-go-lint
+      branches:
+        - ^master$
+        - ^release-1.*$
+      path_alias: istio.io/client-go
+      decorate: true
+      always_run: true
+      spec:
+        containers:
```
Comment (howardjohn) [inline prow/cluster/jobs/istio/client-go/istio-client-go.yaml:13]: do we need resource request and limits specified?

**Comment by geeknoid** [comment]
@howardjohn Yes, we should be using the generator tool for these and many others. We just need to add support to specify the build image.

And regarding the resource values, this is all purely cut & paste from other existing jobs. I don't know how to access how much these jobs consume in order to fine tune. Ideally, I suppose we could have a job to check logs and automatically create a PR to fine tune these values :-)

**Comment by howardjohn** [comment]
that would be cool. I forget what happens if you don't specify requests or
limits, we will need to make sure we don't starve the nodes. All these jobs
are pretty light anyways and take a minute or 2 so I'm not too concerned

**Pull request merged**